src/main/java/datawave/query/util/DateFrequencyValue.java

```diff
@@ -0,0 +1,162 @@
+ * maximum length for a 10 year capture of 14720 bytes.
+ */
+
+public class DateFrequencyValue {
+    
+    private static final Logger log = LoggerFactory.getLogger(DateFrequencyValue.class);
+    
+    public static final int DAYS_IN_LEAP_YEAR = 366;
+    private static int NUM_YEAR_BYTES = 4;
+    private static int NUM_FREQUENCY_BYTES = DAYS_IN_LEAP_YEAR * 4;
+    private static int NUM_BYTES_PER_FREQ_VALUE = 4;
+    
+    public DateFrequencyValue() {}
+    
+    /**
+     *
+     * @param dateToFrequencyValueMap
+     *            the keys should be dates in yyyyMMdd format
+     * @return Value the value to store in accumulo
+     */
+    public Value serialize(TreeMap<YearMonthDay,Frequency> dateToFrequencyValueMap) {
+        
+        Value serializedMap;
+        int year, presentYear = 0;
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        byte[] dayFrequencies = new byte[NUM_FREQUENCY_BYTES];
+        Arrays.fill(dayFrequencies, (byte) 0);
+        
+        for (Map.Entry<YearMonthDay,Frequency> dateFrequencyEntry : dateToFrequencyValueMap.entrySet()) {
+            year = dateFrequencyEntry.getKey().year;
+            if (year != presentYear) {
+                
+                if (presentYear != 0) {
+                    try {
+                        baos.write(dayFrequencies);
+                    } catch (IOException ioException) {
+                        log.error("Did not write out the frequence properly ", ioException);
+                    }
+                }
+                
+                presentYear = year;
+                
+                try {
+                    baos.write(Base256Compression.numToBytes(presentYear));
+                    Arrays.fill(dayFrequencies, (byte) 0);
+                } catch (IOException ioException) {
+                    log.error("Could not convert the year or the first ordinal (julian) to bytes ", ioException);
+                }
+            }
+            
+            putFrequencyBytesInByteArray(dateFrequencyEntry, dayFrequencies);
+            
+            if (log.isTraceEnabled())
+                log.trace(dateFrequencyEntry.getKey().toString());
+            
+        }
+        
+        if (dayFrequencies != null) {
+            try {
+                baos.write(dayFrequencies);
+            } catch (IOException ioException) {
+                log.error("Did not write out the frequence properly ", ioException);
+            }
+        }
+        
+        serializedMap = new Value(baos.toByteArray());
+        
+        return serializedMap;
+    }
+    
+    private void putFrequencyBytesInByteArray(Map.Entry<YearMonthDay,Frequency> entry, byte[] dayFrequencies) {
+        // This will always be 4 bytes now even if null it will get compressed later on.
+        byte[] frequency = Base256Compression.numToBytes(entry.getValue().value);
```

Comment:
Again, this method can write directly to the array instead of allocating a new 4 byte array every time.

Reply:
All that this function is doing is calling the numToBytes function which returns the 4 byte frequency array.  The 4 byte "frequency" array is then copied byte by byte into dayFrequencies.  Copying is not "allocating" and is fast.  Not seeing the problem here.  This function only gets called when there is a Day-Frequency in the TreeMap.   The creation of the 4 byte frequency array is going to happen no matter what by the call to "numToBytes".  

The dayFrequencies array gets written out to the byte stream when there is a year change or when the TreeMap iteration is over.   

I am still trying to see how directly writing to the ByteArrayOutputSteam piecemeal in this function is more efficient than doing it all at once on line line 52 with  baos.write(dayFrequencies);

I will keep looking at it anyway till I see where you are going.